spec.html

```diff
@@ -32006,7 +32006,7 @@
           1. Let _O_ be the *this* value.
           1. Perform ? ValidateTypedArray(_O_).
           1. Let _value_ be ? ToNumber(_value_).
-          1. Let _len_ be ? _O_.[[ArrayLength]].
+          1. Let _len_ be _O_.[[ArrayLength]].
```

Comment:
The suggestion above - inverting the order of steps - allows the implementation to define what is done to `[[ArrayLength]]` or if it should still return a fixed value.

See that `TypedArray#length` returns 0 if it has a detached buffer:

```
If IsDetachedBuffer(buffer) is true, return 0.
Let length be O.[[ArrayLength]].
```

[ValidateAtomicAccess](https://tc39.github.io/ecma262/#sec-validateatomicaccess) is the other case in the specs we can verify [[ArrayLength]] being accessed after an observable step where we do not check for a detached buffer.

```
Let accessIndex be ? ToIndex(requestIndex).
Let length be typedArray.[[ArrayLength]].
```